RNA_folding.py

```diff
@@ -0,0 +1,329 @@
+            zeros = 'Null:' + str(stem)
+            cqm.add_variable(zeros, 'BINARY')
+            cqm.add_discrete(substems + [zeros], stem)
+
+    for stem1, stem2 in combinations(stem_dict.keys(), 2):
+        # Check maximal stems first.
+        if check_overlap(stem1, stem2):
+            # If maximal stems overlap, compare list of smaller stems.
+            for stem_pair in product(stem_dict[stem1], stem_dict[stem2]):
+                if check_overlap(stem_pair[0], stem_pair[1]):
+                    cqm.add_constraint(dimod.quicksum([dimod.Binary(stem) for stem in stem_pair]) <= 1)
+
+    return cqm
+
+
+def process_cqm_solution(sample_set, verbose=True):
+    """ Processes samples from solution and prints relevant information.
+
+    Prints information about the best feasible solution and returns a list of stems contained in solution.
+    Returns solution as a list of stems rather than a binary string.
+
+    Args:
+        sample_set: dimod sampleset element.
+        verbose: Boolean indicating if function should print additional information.
+
+    Returns: List of stems included in optimal solution, encoded as 4-tuples.
+    """
+
+    # Filter for feasibility
+    feasible_samples = sample_set.filter(lambda s: s.is_feasible)
+    # Check that feasible example exists
+    if not feasible_samples:
+        print('\033[93m' + '\nWarning! All solutions infeasible. You may need to try again.')
+        return None
+    # Extract best feasible sample
+    solution = feasible_samples.first
+
+    print('Best Energy:', solution.energy)
+
+    # extract stems with a positive indicator variable.
+    bonded_stems = [stem for stem, val in solution.sample.items() if val == 1 and type(stem) == tuple]
+
+    print('\n# stems in best solution:', len(bonded_stems))
+    print('Stems in best solution:', *bonded_stems)
+
+    if verbose:
+        print('\n# variables (stems):', len(solution[0].keys()))
+
+        # find psuedoknots using product instead of combinations allows for short asymmetric checks.
+        pseudoknots = [(stem1, stem2) for [stem1, stem2] in product(bonded_stems, bonded_stems)
+                       if stem1[1] < stem2[0] and stem2[1] < stem1[2] and stem1[3] < stem2[2]]
+
+        print('\n# pseudoknots in best solution:', len(pseudoknots))
+        if pseudoknots:
+            print('Pseudoknots:', *pseudoknots)
+
+    return bonded_stems
+
+
+# Create command line functionality.
+DEFAULT_PATH = join(dirname(__file__), 'RNA_text_files', 'TMGMV_UPD-PK1.txt')
+
+
+@click.command(help='Solve an instance of the RNA folding problem using '
+                    'LeapHybridCQMSampler.')
+@click.option('--path', type=click.Path(), default=DEFAULT_PATH,
+              help=f'Path to problem file.  Default is {DEFAULT_PATH!r}')
+@click.option('--verbose', is_flag=True, default=True)
+@click.option('--min_stem', type=click.INT, default=50)
+@click.option('--min_loop', type=click.INT, default=2)
```

Comment:
It's minor, but conventionally we use dashes instead of underscores here.  If nothing else, it's easier to type.  (They'll get converted to dashes for the actual variable names internally.)